### PR TITLE
[CU-9phw06] DaskCluster: start workers in parallel

### PR DESCRIPTION
**Pull request by wreis**

### CODE REVIEW

**Comment by hhuuggoo** [comment]
Task linked: [CU-9phw06 start initial dask workers when we start the scheduler?](https://app.clickup.com/t/9phw06)

File: daskdev-sat/app.py [hunk 1]
```diff
@@ -50,10 +55,39 @@ def make_cluster(n_workers):
...
+            )
+            self.scheduler.address = scheduler_address
+            self.scheduler_comm = dask_rpc(
+                scheduler_address,
+                connection_args=self.security.get_connection_args("client"),
+            )
+            self.pod_template.spec.containers[0].env.append(
+                kubernetes.client.V1EnvVar(
+                    name="SATURN_DASK_SCHEDULER_ADDRESS", value=scheduler_address
+                )
```
Comment (wreis) [inline daskdev-sat/app.py:83]: This the part I'm not entirely happy, but I fail to see another way.

dask-kubernetes core sets worker spec like that by [using `DASK_SCHEDULER_ADDRESS` env var ](https://github.com/dask/dask-kubernetes/blob/fdb77c0586fc83fabe3eb3059dd6f6797db0a2fd/dask_kubernetes/core.py#L132)with value set to `self.scheduler` which is passed by dask-distributed [`SpecCluster` package with `self.scheduler.address`](https://github.com/dask/distributed/blob/f761dd21e8cbba62273ab850f5f83e8a45a2fffa/distributed/deploy/spec.py#L348), however for some reason that is getting `None` once worker pods are started.

Comment (jameslamb) [reply daskdev-sat/app.py:83]: hmmmm ok I think I understand, that makes sense

**Comment by wreis** [comment]
> Is it possible to just contribute this directly to dask-kubernetes, or is there something very Saturn-specific about what we're doing?

Well, if we figure out a way to reuse the env var there instead of setting our own, then yeah we could push it out to dask-kubernetes (after it's been tested by us and merged for a while without any big issue).

**Comment by jameslamb** [comment]
> > Is it possible to just contribute this directly to dask-kubernetes, or is there something very Saturn-specific about what we're doing?
> 
> Well, if we figure out a way to reuse the env var there instead of setting our own, then yeah we could push it out to dask-kubernetes (after it's been tested by us and merged for a while without any big issue).

ok, got it. I want to hear if @jsignell has any ideas for alternatives to this issue you faced: https://github.com/saturncloud/images/pull/58#discussion_r480208762. My biggest concern with this PR is that it's exposing our application to a lot more of the `dask-kubernetes` API than the current state, which means there is a greater risk of this breaking as `dask-kubernetes` evolves. Maybe it's unavoidable and we have to do that...I think @jsignell will know better than me

**Comment by jsignell** [comment]
I agree that we are taking on a lot more of dask-kubernetes, but we already do a fair amount of that in dask-saturn so I'm not too worried about it. If we find that it works well, we can suggest a PR upstream to start dask workers at the same time as the scheduler.

**Comment by jameslamb** [comment]
alright, sounds good to me. Thanks

**Pull request merged**